### PR TITLE
Open FxA/Sync Prefs

### LINKED ISSUE #92: Open FxA/Sync Account Details
As a user, I want to be taken to where I can manage my account and sync information.

### Acceptance Criteria
* Should be able to navigate to Account details from the drop down menu as created in #72 - to either sign in if not already or to manage my existing signed in state - about:preferences#sync

### Implementation Details
* Add a "service" parameter for telemetry noting Lockbox is the source

### Visual Design
* https://mozilla.invisionapp.com/share/ENOL84959AK#/screens/338158451

child of #38

### PR DESCRIPTION
**Pull request by linuxwolf**
Fixes #92 

Adds a WebExt API for opening FxA/Sync Preferences within Firefox, so that the user menu items for "Account" and "Sign in to Sync" do something.

The work for #61 depends on at least the webext API, and probably also the helper added herein.

## Testing and Review Notes

* Integration tests verify the webext API opens "about:preferences...#sync".
* Clicking "Account" or "Sign in to Sync" from the user menu on the management page should open the FxA/Sync preferences.

The integration tests are kind of "brute force" walking the open windows/tabs looking for a matching URL.  There might be a better way to do this, but it's not obvious.

## Screenshots or Videos

* "Sign in to Sync" menu item (no account yet):
desktop-92 sign-in-menu

* "Account" menu item (signed into Sync):
desktop-92 account-menu

## To Do

- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] add integration tests
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

**Comment by linuxwolf** [comment]
@6a68 / @lmorchard / @meandavejustice I could use some help completing the integration tests.  Namely, the best way to verify the preferences have actually opened.

**Comment by 6a68** [comment]
> @6a68 / @lmorchard / @meandavejustice I could use some help completing the integration tests. Namely, the best way to verify the preferences have actually opened.

Eh, I think you're probably fine. The approach used in the [existing sync tests in-tree](https://searchfox.org/mozilla-central/source/browser/components/customizableui/test/browser_synced_tabs_menu.js#115) is to get the window via `gBrowser` tricks, then check its URL via comparing `gBrowser.currentURI.spec` to an `about:preferences` string. You could replace the `found` boolean with the full URL of the page, to ensure the correct panel is opened, but that seems optional to me.

File: src/experiments/sync/api.js [hunk 1]
```diff
@@ -45,6 +45,16 @@ const getProfileInfo = async () => {
   return profileInfo;
 };
 
+const openPrefs = async (origin, entrypoint) => {
+  const win = Services.wm.getMostRecentWindow("navigator:browser");
+
+  win.openPreferences("paneSync", { origin, urlParams: { entrypoint } });
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/experiments/sync/api.js]: It didn't look like `gSync` is exported ...

[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/experiments/sync/api.js]: nm, I see it now.

[the remark this answers is not in the document]
Comment (linuxwolf) [reply src/experiments/sync/api.js]: (after IRL chat) and maybe nevermind again; `gSync` needs a global `window`.  We can revisit if the current approach after "external" review.

**Pull request merged**